skbuild/cmaker.py

```diff
@@ -381,10 +382,11 @@ def get_python_library(python_version):
             candidates = (
                 os.path.join(
                     libdir,
-                    ''.join((pre, 'python', ver, abi, ext))
+                    ''.join((pre, impl, ver, abi, ext))
                 )
-                for (pre, ext, ver, abi) in itertools.product(
+                for (pre, impl, ext, ver, abi) in itertools.product(
                     candidate_lib_prefixes,
+                    ('python', 'pypy', 'pypy-c'),
```

Comment:
What do you think of using a global dictionary and lookup the implementation based on the return value of `platform.python_implementation()` ? 

It would also be nice to setup CI for testing scikit-build with PyPy ?  Is it something you could help  with ? May be extending the existing CircleCI or azure-pipelines configuration ?

Reply:
Which one would you prefer, circleci or azure?

Reply:
If you end up using a Docker image, leveraging CircleCI may make more sense as you can easily specify a container to run the test with.

May be you could look into using one of these images: https://marketplace.automic.com/details/pypy-official-docker-image ?

On linux, I wonder if requirements for pypy are similar to the one for manylinux ? (meaning a compiling environment with a modern compiler but dependency on an older  glibc. See [pep-513](https://www.python.org/dev/peps/pep-0513) and [pep-571](https://www.python.org/dev/peps/pep-0571))

After the details are ironed out, I also wonder if (similarly to what is done for manylinux) integrating an image in [dockcross](https://github.com/dockcross/dockcross) would be relevant. This would provide one (or multiple images) with a readily available compiling env.